pkg/exec/cli_linux.go

```diff
@@ -7,18 +7,17 @@ import (
 )
 
 func performOSChecks(spyName string) error {
-	if disableLinuxChecks {
-		return nil
-	}
-	if spyName == "ebpfspy" {
+	var err error
+	switch {
+	case disableLinuxChecks:
+	case spyName == "dotnetspy":
+	case spyName == "ebpfspy":
 		if !isRoot() {
-			return errors.New("when using eBPF you're required to run the agent with sudo")
-		}
-	} else {
-		if !caps.HasSysPtraceCap() {
-			return errors.New("if you're running pyroscope in a Docker container, add --cap-add=sys_ptrace. " +
-				"See our Docker Guide for more information: https://pyroscope.io/docs/docker-guide")
+			err = errors.New("when using eBPF you're required to run the agent with sudo")
```

Comment:
another nood question:  

previously for dotnetspy,  it did the sys ptrace cap check, but did not check root,  now seems like the condition checks for dotnetspy actually change ?  and for dot net, I am seeing those lines in the Dockerfile as well:

`
RUN adduser --disabled-password --gecos --quiet pyroscope
USER pyroscope
CMD ["pyroscope", "exec", "dotnet", "/dotnet/example.dll"]
`

so looks like it is going to run the agent as user pyroscope , not root?   will it still be failing this performOSChecks ?

Reply:
ah @kolesnikovae , actually I had to admit it is not because the diffs are unreadable, but because my knowledge about go lang is weak, just found out that " Go switch statements do not fall through by default. ".   
Now I get it!  Thanks for the explanation!